### ISSUE TITLE
FancyEqMixin should cooperate with equality implementations from other types

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2944]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2944|
|Type|enhancement|
|Created|2007-12-10 16:16:27Z|
|Branch|https://github.com/twisted/twisted/tree/fancy-eq-mixin-2944|

Python's ```NotImplemented``` mechanism allows both operands to contribute to the result of binary comparisons.  ```FancyEqMixin``` should take advantage of this feature.

<details><summary>Searchable metadata</summary>

```
trac-id__2944 2944
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__branches_fancy_eq_mixin_2944 branches/fancy-eq-mixin-2944
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1197303387000000 1197303387000000
changetime__1202991658000000 1202991658000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [22544]) Branching to 'fancy-eq-mixin-2944'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Perhaps that is an adequate resolution.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [22546]) Merge fancy-eq-mixin-2944

Author: exarkun
Reviewer: therve
Fixes [#2944](https://github.com/twisted/twisted/issues/2944)

Add more complete unit tests for twisted.python.util.FancyEqMixin and
change FancyEqMixin so that it cooperates with the equality implementations
of other types.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
